### PR TITLE
kernel/timer: Fix security vulnerability #53 - raw kernel pointer handles

### PR DESCRIPTION
**Pull request by vivek1-j**
POSIX timer APIs (timer_delete, timer_settime, timer_gettime) previously accepted arbitrary timerid values without validating that the handle belongs to the allocated timer list. This security vulnerability allowed forged, stale, and crafted handles to be used, leading to memory corruption, use-after-free, and double-free in kernel code.

This PR adds timer_gethandle() — a kernel-internal validation API that verifies timer handles by checking g_alloctimers list membership and PT_FLAGS_INUSE flag. All POSIX timer APIs are updated to use this validation, properly rejecting invalid/stale/forged handles with EINVAL.

**Changes**
Commit 1: apps/testcase: add timer handle security validation test cases
Add kernel TC tests for timer_gethandle() validation (null, valid, forged, after-delete)
Add forged handle tests for timer_delete(), timer_settime(), timer_gettime()
Add ITC tests for double-delete and forged handle rejection
Add TC_KERNEL_TIMER_GETHANDLE Kconfig option
Guard timer_gethandle direct tests with #ifndef CONFIG_BUILD_PROTECTED
Commit 2: os/kernel: add timer_gethandle() to prevent forged handle attacks
Add timer_gethandle() in timer_initialize.c that validates handles via g_alloctimers list iteration
Update timer_delete() to use timer_gethandle() instead of direct pointer cast
Update timer_settime() and timer_gettime() to use timer_gethandle() instead of weak PT_ISVALID() check
Remove PT_ISVALID() macro, replace with timer_gethandle() declaration in timer.h
Fix test_timer_initialize() in os_api_test to call timer_delete() before re-initializing
**Testing Summary**
Table 1: Test Results Comparison
| S.No. | Test (Function Name) | Before (Vulnerable) | After (Fixed) |
|-----|------|---------------------|--------------|
| 1 | tc_timer_timer_getoverrun | PASS | PASS |
| 2 | tc_timer_delete_forged_handle | FAIL | PASS |
| 3 | tc_timer_settime_forged_handle | PASS | PASS |
| 4 | tc_timer_gettime_forged_handle |	PASS | PASS |
| 5 | tc_timer_timer_set_get_time | PASS | PASS |
| 6 | tc_timer_timer_initialize | PASS | PASS |
| 7 | itc_timer_delete_n_after_delete | FAIL | PASS |
| 8 | itc_timer_delete_forged_handle | FAIL | PASS |
| 9 | itc_timer_set_time_n_after_delete | PASS | PASS |
| 10 | itc_timer_get_time_n_after_delete | PASS | PASS |

Table 2: Vulnerability Analysis
| Vulnerability | Before (Exploitable/Blocked - Return Value) | After (Exploitable/Blocked - Return Value) |
|-----|-----|-----|
| Forged handle passed to timer_delete() | Exploitable — Returns OK (0), arbitrary memory freed | Blocked — Returns ERROR (-1), errno=EINVAL |
| Stale/double-deleted handle passed to timer_delete() | Exploitable — Returns OK (0), double-free occurs | Blocked — Returns ERROR (-1), errno=EINVAL |
| Forged handle passed to timer_settime() | Weakly Blocked — Returns ERROR (-1) via PT_ISVALID(), but only checks flag bit at pointer address (bypassable with crafted pointer) | Blocked — Returns ERROR (-1), errno=EINVAL via timer_gethandle() list membership validation |
| Forged handle passed to timer_gettime() | Weakly Blocked — Returns ERROR (-1) via PT_ISVALID(), but only checks flag bit at pointer address (bypassable with crafted pointer) | Blocked — Returns ERROR (-1), errno=EINVAL via timer_gethandle() list membership validation |

**Impact**
No breaking changes for legitimate timer usage — valid handles continue to work as before
Protected build compatible — forged handle tests for public POSIX APIs work through syscalls; timer_gethandle direct tests are guarded with #ifndef CONFIG_BUILD_PROTECTED
All 10 kernel timer TCs pass after the fix (previously 7 pass / 3 fail)

Signed-Off by: Vivek Jain (vivek1.j@samsung.com)

### CODE REVIEW

File: os/kernel/timer/timer_release.c [hunk 1]
```diff
@@ -142,6 +142,8 @@ static inline void timer_free(struct posix_timer_s *timer)
 
 int timer_release(FAR struct posix_timer_s *timer)
 {
+	irqstate_t flags;
```
Comment (seokhun-eom24) [inline os/kernel/timer/timer_release.c]: It's not used.

File: apps/examples/testcase/le_tc/kernel/Kconfig [hunk 2]
```diff
@@ -182,6 +182,14 @@ config TC_KERNEL_TIMER
 	bool "Timer"
 	default n
 
+config TC_KERNEL_TIMER_GETHANDLE
```
Comment (seokhun-eom24) [inline apps/examples/testcase/le_tc/kernel/Kconfig]: It's not used.
Do we need this? Because we don't have config for `timer_gethandle`.

File: os/kernel/timer/timer_release.c (hunk 1 above)
Comment (vivek1-j) [reply os/kernel/timer/timer_release.c]: It is not required, looks like residual code from previous dev iteration. It is now removed.

File: apps/examples/testcase/le_tc/kernel/Kconfig (hunk 2 above)
Comment (vivek1-j) [reply apps/examples/testcase/le_tc/kernel/Kconfig]: It is not required, looks like residual code from previous dev iteration. It is now removed.
It was added to protect against linker error for timer_gethandle TC's in protected build, now, I am directly using CONFIG_BUILD_PROTECTED.

**Review by seokhun-eom24 — APPROVED** [review]
